### PR TITLE
fix: useKeyPress 绑定 keyFilter 冲突(#1452)

### PR DESCRIPTION
**Pull request by SuperHuangXu**
[[中文版模板 / Chinese template](https://github.com/alibaba/hooks/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

File: packages/hooks/src/useKeyPress/index.ts [hunk 1]
```diff
@@ -147,10 +147,21 @@ function genFilterKey(event: KeyboardEvent, keyFilter: keyType) {
   // 字符串依次判断是否有组合键
   const genArr = keyFilter.split('.');
   let genLen = 0;
+  // 是否点击了修饰键
```
Comment (brickspert) [inline packages/hooks/src/useKeyPress/index.ts]: 这边的变更应该是有问题的。比如对于数字的监听

```ts
  useKeyPress(40, () => {
    setCounter((s) => s - 1);
  });
```

这种情况点击 `ctrl + 40` 仍然会触发执行。因为上面 `typeof keyFilter === 'number'` 没有处理。

Comment (SuperHuangXu) [reply packages/hooks/src/useKeyPress/index.ts]: 麻烦再看下～

File: packages/hooks/src/useKeyPress/index.ts [hunk 2]
```diff
@@ -151,6 +160,9 @@ function genFilterKey(event: KeyboardEvent, keyFilter: keyType) {
   for (const key of genArr) {
     // 组合键
     const genModifier = modifierKey[key];
+    if (pressModifier && genArr.length === 1) {
```
Comment (brickspert) [inline packages/hooks/src/useKeyPress/index.ts]: 这里的变更考虑的不全面，我举个例子：

比如我想监听 `ctrl.a`，但用户点击了 `ctrl.shift.a`，理论上不应该监听的。

但目前的代码变更并没有处理这种情况。

Comment (SuperHuangXu) [reply packages/hooks/src/useKeyPress/index.ts]: 我把 `git log`  弄太乱了，重新提交了个 [pr](https://github.com/alibaba/hooks/pull/1466)

**Pull request closed without merging**